sktime/clustering/partitioning/_time_series_k_partition.py

```diff
@@ -0,0 +1,250 @@
+        "wddtw": wddtw_distance,
+        "lcss": lcss_distance,
+        "erp": erp_distance,
+        "msm": msm_distance,
+        "twe": twe_distance,
+    }
+
+    # "k_means_plus_plus": KMeansPlusPlusCenterInitializer,
+    __init_algorithms: Init_Algo_Dict = {
+        "forgy": ForgyCenterInitializer,
+    }
+
+    def __init__(
+        self,
+        n_clusters: int = 8,
+        init_algorithm: Init_Algo = "forgy",
+        max_iter: int = 300,
+        verbose: bool = False,
+        metric: Metric_Parameter = "dtw",
+    ):
+        """
+        Constructor for time_series_k_partition clusterer
+
+        Parameters
+        ----------
+            n_clusters: int, default = 8
+                The number of clusters to form as the number of
+                centroids to generate.
+
+            init_algorithm: Init_Algo, default = forgy
+                Algorithm that is used to initialise the cluster
+                centers. str options are "forgy", "random" or
+                "k-means++". If using custom center init algorithm
+                then must be of type Init_Algo
+
+            max_iter: int, default = 300
+                Maximum number of iterations of time series k means
+                for a single run.
+
+            verbose: bool, default = False
+                Verbosity mode.
+
+            metric: Metric_Parameter, default = None
+                The distance metric that is used to calculate the
+                distance between points.
+
+        """
+        super(TimeSeriesKPartition, self).__init__()
+
+        self.n_clusters = n_clusters
+        self.max_iter = max_iter
+        self.verbose = verbose
+        self.metric = metric
+        self.init_algorithm = init_algorithm
+
+        self.__centers: Numpy_Array = None
+        self._init_algorithm = None
+        self._metric = None
+
+    def _fit(self, X: Numpy_Array) -> None:
+        """
+        Method that is used to fit the time series k
+        partition model on dataset X
+
+        Parameters
+        ----------
+        X: Numpy_Array
+            Numpy array of series to train the model on
+
+        Returns
+        -------
+        self
+            Fitted estimator
+        """
+        center_algo: BaseClusterCenterInitializer = self._init_algorithm(
+            X, self.n_clusters
+        )
+        self.__centers = center_algo.initialize_centers()
+
+        for _ in range(self.max_iter):
+            self.__update_centers(X)
+
+    def _predict(self, X: Numpy_Array) -> Numpy_Array:
+        """
+        Method used to perform a prediction from the trained
+        time series k partition clustering algorithm
+
+        Parameters
+        ----------
+        X: Numpy_Array
+            Numpy_Array containing the time series to
+            predict clusters for
+
+        Returns
+        -------
+        Numpy_Array
+            Index of the cluster each sample belongs to
+        """
+
+        if self.__centers is None:
```

Comment:
This should be handled via `self.check_is_fitted()`. If you still want to raise a separate exception here, please make it more specific, e.g. `AttributeError`